src/backend/cuda/kernel/ireduce.hpp

```diff
@@ -27,6 +27,9 @@ namespace kernel
     static double __host__ __device__ cabs(const char in) { return (double)(in > 0); }
     static double __host__ __device__ cabs(const cfloat &in) { return (double)abs(in); }
     static double __host__ __device__ cabs(const cdouble &in) { return (double)abs(in); }
+    template<typename T> __host__ __device__ static bool isNan(T in) { return in != in; }
+    static bool __host__ __device__ isNan(const cfloat &in) { return in.x != in.x || in.y != in.y; }
+    static bool __host__ __device__ isNan(const cdouble &in) { return in.x != in.x || in.y != in.y; }
```

Comment:
Can you specialize this function so that it is a template? It should look something like this.

```
template<> __host__ __device__ isNan<cfloat>(const cfloat &in) {...}
template<> __host__ __device__ isNan<cdouble>(const cdouble &in) {...}
```

Reply:
Just read up on template specialization. Haha I have a lot to learn in C++, thanks for this tip! I went ahead and did the same too for `cabs()`